### ISSUE TITLE
add missing documentation of use cases for workflowrun states

### ISSUE THREAD

**Issue by tomsit-ionos** [github/docs#34987]
### Code of Conduct

- [X] I have read and agree to the GitHub Docs project's [Code of Conduct](https://github.com/github/docs/blob/main/.github/CODE_OF_CONDUCT.md)

### What article on docs.github.com is affected?

https://docs.github.com/en/rest/actions/workflow-runs?apiVersion=2022-11-28#list-workflow-runs-for-a-repository

but should be linked/included on all other pages that are concerned with the WF run status

### What part(s) of the article would you like to see updated?

most of the statuses are self explanatory but there is a gray area.

1. Plz provide a description/use case/trigger for each of the statuses, preferably with a state diagram.
In particular i would like to know:

    the meaning of these: `neutral, stale`
    the difference/relation of these: `queued, requested, waiting, pending, action_required`

3. Delineate which value belongs to which property as per the above doc:
![image](https://github.com/user-attachments/assets/1eda6497-d222-402f-932c-eb5a2966c172)

### Additional information

PS: this is a rehash from here: https://github.com/orgs/community/discussions/70540#discussioncomment-10978598

_No response_

**Comment by tomsit-ionos** [comment]
btw: i just found this case: ` "conclusion": "startup_failure"`

**Comment by nguyenalex836** [comment]
@tomsit-ionos Thank you for raising this issue! I'll get this triaged for review :sparkles: Our team will provide feedback regarding the best next steps for this issue - thanks for your patience! 💛

**Comment by tomsit-ionos** [comment]
also it is not quite obvious why the workflow can transition like so:

`queued -> in_progress -> queued -> completed`

**Comment by xxx2585** [comment]
[Uploading Getting-Started.md…]()

**Comment by nguyenalex836** [comment]
@tomsit-ionos Thank you for your patience while our SME team reviewed! Our SME team agrees it would be valuable to have documentation covering these statuses and conclusions.

In terms of ideal location for this new content, we would prefer creating a new reference type of article to [Managing workflow runs](https://docs.github.com/en/actions/managing-workflow-runs-and-deployments/managing-workflow-runs) that defines these, and then link to that article from the API docs.

I've added the `help wanted` label so that you, or anyone else with knowledge of this area, may work on adding this new content 💛

**Comment by 55-x** [comment]
Is this still being worked on?

**Comment by cceneag** [comment]
是的我正在參加github的課程有什麼特別還是建議嗎？

ozu ***@***.***>於 2025年6月3日 週二，下午9:01寫道：

**Comment by ayushmank** [comment]
hi @tomsit-ionos is this issue still opens 
please let me know
i would like to contribute in this

**Comment by tomsit-ionos** [comment]
Yes this seems to be still open - at least looking at the API that I referenced in the description , still doesn't have an easily accessible link or description to the meaning of the states and their transitions

**Comment by ayushmank** [comment]
Hi @tomsit-ionos please have a look to this pr [#41379 ] when you have free time 
thanks !!

**Comment by subatoi** [comment]
Hi @tomsit-ionos—

I apologise for the late conclusion here. In fact, this page has existed since early last year: https://docs.github.com/en/pull-requests/reference/status-checks. That to say, you raising an issue here in October of 2024 was entirely valid, and I regret that you weren't informed sooner.

I do see the potential value in adding a link to the REST API docs that you referenced; however, we wouldn't be able to accept a contribution to do that in this repo. I will, instead, pass that feedback on to the relevant team, but since I can't guarantee a timeframe for a response from them and the main part of this issue's problem is resolved, I'm going to go ahead and close this out.

Apologies again for the delay, and thank you very much for your contribution and interest in the GitHub docs 🙇
